docs/batch_inference_with_mms.md

```diff
@@ -0,0 +1,335 @@
+        "class": "n02124075 Egyptian cat"
+      },
+      {
+        "probability": 0.00837081391364336,
+        "class": "n02127052 lynx, catamount"
+      },
+      {
+        "probability": 0.0006728120497427881,
+        "class": "n02129604 tiger, Panthera tigris"
+      }
+    ```
+    
+* Now that we have the service up and running, we could run performance tests with the same kitten image as follows.
+```text
+$ ab -k -l -n 10000 -c 1000 -T "image/jpeg" -p kitten.jpg localhost:8080/predictions/resnet-152
+```
+To run this, we would need to install "apache bench" first. Since we were running this on a ubuntu host, we installed this as follows
+```bash
+$ sudo apt-get udpate && sudo apt-get install apache2-utils
+``` 
+
+## Performance benchmarking
+We benchmarked MMS with batch-inference enabled Resnet-152 on a P3.8xlarge instance, which is a AWS provided GPU EC2 instance. 
+We ran MMS in our [GPU container](https://hub.docker.com/r/awsdeeplearningteam/mxnet-model-server/tags) which hosted the above resnet-152 model. 
+We ran the tests for batch sizes 1, 8 and captured results. 
+We saw a significant gain in throughput and also saw that the GPU resources were utilized more optimally. Attached is the graph showing the throughput gains. The 
+experiment was done with the following configuration
+
+```bash
+$ cat config.properties
+model_store=/opt/ml/model
+inference_address=http://0.0.0.0:8080
+management_address=http://0.0.0.0:8081
+number_of_netty_threads=32
+job_queue_size=1000
+async_logging=true
+
+# To load the model run the following command
+$ curl -X POST "localhost:81/models?url=https://s3.amazonaws.com/model-server/model_archive_1.0/examples/resnet-152-batching/resnet-152.mar&batch_size=8&max_batch_delay=50&initial_workers=8"
+
+
+# Apache bench performance test command
+$ ab -k -l -n 10000 -c 1000 -T "image/jpeg" -p kitten.jpg localhost:8080/predictions/resnet-152
+```
+We set the `batch_size` at 1, 8 and 16 and captured the results of `ab` command by setting the `-c` option, or concurrency, at 10, 50, 100, 500 and 1000 for each of the `batch_size`s.
+As seen from the command, the number of workers was set at 8 (2 per GPU), and `max_batch_delay` was set at 50 ms.  
+
+![Graph](images/throughput.png)
+
+As we can see from the above diagram, batching is not a one-size-fits-all solution. For example, when the rate of requests received at MMS is lower, look at 10 concurrent clients,
+the batch size of 1 out performs batch size of 8 or 16. This is because, some requests would wait until the 50ms timeout occured before getting scheduled to be processed 
+by the model handlers. Whereas, by definition, batch size of 1 wouldn't wait for this timeout to occur and the model handler is given the request to handle as and when
+the requests come in.
+
+To understand this behavior a little better, we kept the batch-size at 8 and number of workers at 8 and varied the `max_batch_delay` and ran `ab`. This showed that as 
```

Comment:
I suggest to provide some more test setup data here. Namely, what was the concurrency level here, as with higher concurrency we might see a different picture for same `max_batch_delay` values. Maybe it even makes sense to supply couple metrics here - all on one graph with fixed `num_workers` and `batch-size` but varying _-c_ param for `ab` command. So that each line would represent different _-c_ value. I expect with higher values for _-c_ we will see bigger TPS at bigger `max_batch_delay`.

Reply:
Bump to this feedback.

Reply:
I plotted two graphs in this document.
1. Varying batch_size and concurrency with constant delay
2. Varying delay and concurrency with constant batch_size

These two graphs show that not always gives best performance, but infact this gives good performance when you have high traffic. 

With regards to the constant workers, I tested this with 8 and 16 workers but didn't see a significant change in numbers. So I updated this graph results from 8 workers.